### PR TITLE
Display last successful sync

### LINKED ISSUE #1624: Menu: Sync: Last sync is showing the last attempt to sync but the last "successful" sync
## What went wrong? 😲
 

Last sync is showing the last attempt to sync not the last "successful" sync

Screen Shot 2023-04-27 at 1 38 34 PM

Screen Shot 2023-04-27 at 1 38 48 PM

## Expected behaviour 🤔

should show the last time sync has been successful

## How to Reproduce 🔨

Steps to reproduce the behaviour:

you should be "unable to connect to server"

### Agreed solution 

Update to have two statuses: Last successful sync, and last attempt to sync

## Your environment 🌱
 - Version 1111
 - Platform windows desktop

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #1624 

# 👩🏻‍💻 What does this PR do? 
 
Adds another node to the sync status, which shows time of the last successful sync. It displays the time that the last successful sync was started - and I was unsure whether that (which is consistent with the other 'last sync' value) is better than showing when the last successful sync was finished (which is likely more useful, but inconsistent). Opinions welcome.

Screenshot 2023-07-31 at 3 08 45 PM

# 🧪 How has/should this change been tested? 

View the sync status. Break sync and try again.

## 📃 Documentation

- [ ] Update screenshots of sync

### CODE REVIEW

File: server/service/src/sync/sync_status/status.rs [hunk 1]
```diff
@@ -217,6 +224,93 @@ fn get_latest_sync_status(ctx: &ServiceContext) -> Result<Option<FullSyncStatus>
...
+            done: pull_remote_progress_done.map(i32_to_u32),
+        }),
+        push: push_started_datetime.map(|started| SyncStatusWithProgress {
+            started,
+            finished: push_finished_datetime,
+            total: push_progress_total.map(i32_to_u32),
+            done: push_progress_done.map(i32_to_u32),
+        }),
+    };
+    Ok(Some(result))
```
Comment (clemens-msupply) [inline server/service/src/sync/sync_status/status.rs:311]: nit: implement `FullSyncStatus::from_sync_row(row: SyncLogRow)` or similar? (it's the same code as for the other query isn't, it?)

**Comment by clemens-msupply** [comment]
Looks good, just some suggestions /questions:

Find it confusing that the "last successful" line pops up when you press "sync" even there wasn't a sync error. I.e. the line shows, it syncs, and then the line disappears. Would it be better to only show it when there was an actual error?

In the screenshot the "3 days ago" shows in a second line. Can't you make the used space a bit wider so that it fits into one line?

**Review by clemens-msupply — APPROVED** [review]

File: server/service/src/sync/sync_status/status.rs (hunk 1 above)
Comment (mark-prins) [reply server/service/src/sync/sync_status/status.rs:311]: it is - and that's much nicer thnks

**Comment by mark-prins** [comment]
🙏  @clemens-msupply for review!
I had the same concern about showing on two rows, and had a quick look. decided that I liked the constrained width and was willing to sacrifice the single row, but with your prompting, will revisit! thanks.

will also check for `in_progress` and only show the successful sync if there is an error.

**Comment by mark-prins** [comment]
updated: 

image

**Pull request merged**